### PR TITLE
WL-3544 : A announcement is public if it is in a public site or is public

### PR DESCRIPTION
**Pull request by nicholaswilson100**
Issue: Annoucements made on .anon site do not appear in public dropdown list

### CODE REVIEW

File: announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java [hunk 1]
```diff
@@ -1738,7 +1738,7 @@ private List getMessagesPublic(Site site, AnnouncementChannel defaultChannel, Fi
 		{
 			AnnouncementMessage aMessage = (AnnouncementMessage) messageList.get(i);
 			String pubview = aMessage.getProperties().getProperty(ResourceProperties.PROP_PUBVIEW);
-			if (pubview != null && Boolean.valueOf(pubview).booleanValue())
+			if ((site.isPubView()) || (pubview != null && Boolean.valueOf(pubview).booleanValue()))
```
Comment (buckett) [inline announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java]: `Site.isPubView()` is acutally if the site should be included in the public directory of sites, not that the site can be publicly accessed. When you grant access in Site Info -> Manage Access you are adding the .anon role to the site.

**Comment by nicholaswilson100** [comment]
Changed site.isPubView() site.hasAnonRole()

**Comment by nicholaswilson100** [comment]
Paired with WL-3544 in kernel

File: announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java [hunk 2]
```diff
@@ -1738,7 +1738,8 @@ private List getMessagesPublic(Site site, AnnouncementChannel defaultChannel, Fi
 		{
 			AnnouncementMessage aMessage = (AnnouncementMessage) messageList.get(i);
 			String pubview = aMessage.getProperties().getProperty(ResourceProperties.PROP_PUBVIEW);
-			if (pubview != null && Boolean.valueOf(pubview).booleanValue())
+			boolean hasAnonRole = site.hasAnonRole();
```
Comment (buckett) [inline announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java]: You probably want todo a security check to work out if the site allows public viewing of announcements passing in the anonymous user and the permission you are wanting to check.

**Comment by nicholaswilson100** [comment]
I've found this method (AnnouncementAction.java, line 4578), do you think it's useful at all here for checking if the site alllows public annoucement viewing?

```
    /**
 * Check if the channel (in effect the site) is public. This is useful for restricting the access options
 * for announcements in a public site.
 * @param channelId The channel ID.
 */
private boolean isChannelPublic(String channelId) {
    return m_securityService.unlock(UserDirectoryService.getAnonymousUser(), AnnouncementService.SECURE_ANNC_READ, channelId);
}  
```

**Comment by buckett** [comment]
That looks to be the sort of thing.

On 27 August 2014 13:59, Nick Wilson notifications@github.com wrote:

> I've found this method (AnnouncementAction.java, line 4578), do you think
> it's useful at all here for checking if the site alllows public annoucement
> viewing?
> 
> ```
> /**
> ```
> - Check if the channel (in effect the site) is public. This is useful for restricting the access options
> - for announcements in a public site.
> - @param channelId The channel ID.
>   */
>   private boolean isChannelPublic(String channelId) {
>   return m_securityService.unlock(UserDirectoryService.getAnonymousUser(), AnnouncementService.SECURE_ANNC_READ, channelId);
>   }
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ox-it/wl-announcement/pull/5#issuecomment-53568267.

## 

  Matthew Buckett, VLE Developer, IT Services, University of Oxford

**Comment by nicholaswilson100** [comment]
Thanks.

**Comment by nicholaswilson100** [comment]
I guess I can probably close https://github.com/ox-it/wl-kernel/pull/44

**Comment by buckett** [comment]
I don't think this one is needed any more is it? ox-it/wl-message#1 is all that's needed.

**Comment by buckett** [comment]
Not needed any more.

**Pull request closed without merging**